streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -299,6 +301,7 @@ public boolean isRunning() {
     private final java.util.function.Consumer<Long> cacheResizer;
 
     private java.util.function.Consumer<Throwable> streamsUncaughtExceptionHandler;
+    private final Map<String, java.util.function.Consumer<Throwable>> topologyExceptionHandlers;
```

Comment:
Do we want to be able to clean these up after we removing a topology? We might not but it might also be a good idea.

Example we remove a topology with a handler and replace one with the same name without a handler or using the non named topology handler

Reply:
Good catch, yes we should clean up after a topology is removed